jbpm-wb-common/jbpm-wb-common-client/src/main/java/org/jbpm/workbench/common/client/dataset/DataSetAwareSelect.java

```diff
@@ -0,0 +1,126 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jbpm.workbench.common.client.dataset;
+
+import javax.enterprise.context.Dependent;
+import javax.enterprise.event.Observes;
+import javax.inject.Inject;
+
+import org.dashbuilder.common.client.error.ClientRuntimeError;
+import org.dashbuilder.dataset.DataSet;
+import org.dashbuilder.dataset.DataSetLookup;
+import org.dashbuilder.dataset.client.DataSetClientServices;
+import org.dashbuilder.dataset.client.DataSetReadyCallback;
+import org.jbpm.workbench.common.client.resources.i18n.Constants;
+import org.jbpm.workbench.df.client.events.DataSetReadyEvent;
+import org.jbpm.workbench.df.client.filter.FilterSettings;
+import org.jbpm.workbench.ks.integration.ConsoleDataSetLookup;
+import org.kie.workbench.common.workbench.client.error.DefaultWorkbenchErrorCallback;
+import org.uberfire.client.views.pfly.widgets.ErrorPopup;
+import org.uberfire.client.views.pfly.widgets.Select;
+
+@Dependent
+public class DataSetAwareSelect {
+
+    @Inject
+    protected DataSetClientServices dataSetClientServices;
+
+    @Inject
+    protected ErrorPopup errorPopup;
+
+    @Inject
+    protected Select select;
+
+    private String valueColumnId;
+    private String textColumnId;
+    private DataSetLookup dataSetLookup;
+    private String tableKey;
+
+    public void setDataSetLookup(DataSetLookup dataSetLookup) {
+        this.dataSetLookup = dataSetLookup;
+    }
+
+    public void setTextColumnId(String textColumnId) {
+        this.textColumnId = textColumnId;
+    }
+
+    public void setValueColumnId(String valueColumnId) {
+        this.valueColumnId = valueColumnId;
+    }
+
+    public void setTableKey(String tableKey) {
+        this.tableKey = tableKey;
+    }
+
+    public void onDataSetReady(@Observes DataSetReadyEvent event) {
+        final FilterSettings filterSettings = event.getFilterSettings();
+        if (filterSettings.getKey().equals(this.tableKey) == false) {
+            return;
+        }
+
+        if (filterSettings.getServerTemplateId() == null || filterSettings.getServerTemplateId().isEmpty()) {
+            removeOptions();
+            return;
+        }
+
+        try {
+            dataSetClientServices.lookupDataSet(ConsoleDataSetLookup.fromInstance(dataSetLookup,
+                                                                                  filterSettings.getServerTemplateId()),
+                                                new DataSetReadyCallback() {
+                                                    @Override
+                                                    public void callback(final DataSet dataSet) {
+                                                        select.refresh(s -> {
+                                                            s.removeAllOptions();
+                                                            for (int i = 0; i < dataSet.getRowCount(); i++) {
```

Comment:
When there aren't process to show, do you think is a good idea disable Selector. I'm saying that because when there's no  process name to show a vertical line appears under the button (the dropdown with no results to show).
Does this make sense? I think we can leave this kind of improvements for later, this comment is just to be sure.